bin/pycbc_convertinjfiletohdf

```diff
@@ -0,0 +1,181 @@
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+""" This program converts standard injection file formats (e.g. XML format
+or the LIGO HDF format) into a PyCBC hdf format that includes a template hash
+used to associate triggers with their template.
+"""
+
+import argparse
+import numpy
+import h5py
+
+import pycbc
+from pycbc.inject import InjectionSet, CBCHDFInjectionSet
+from pycbc.types import float64, float32
+from pycbc.io.record import FieldArray
+
+from pycbc.io.ligolw import LIGOLWContentHandler
+from ligo.lw import utils as ligolw_utils, lsctables
+
+
+# Handlers for alternative injection formats
+
+def legacy_approximant_name(apx):
+    """Convert the old style xml approximant name to a name
+    and phase_order.
+    """
+    apx = str(apx)
+    try:
+        order = sim.GetOrderFromString(apx)
+    except:
+        print("Warning: Could not read phase order from string, using default")
+        order = -1
+    name = sim.GetStringFromApproximant(sim.GetApproximantFromString(apx))
+    return name, order
+
+
+class XMLInjectionSet(object):
+
+    """Reads injections from LIGOLW XML files
+
+    Parameters
+    ----------
+    sim_file : string
+        Path to a LIGOLW XML file containing a SimInspiralTable
+        with injection definitions.
+
+    Attributes
+    ----------
+    indoc
+    table
+    """
+
+    def __init__(self, sim_file):  
+        self.indoc = ligolw_utils.load_filename(
+            sim_file, False, contenthandler=LIGOLWContentHandler)
+        self.table = lsctables.SimInspiralTable.get_table(self.indoc)
+
+    def end_times(self):
+        """Return the end times of all injections"""
+        return [inj.time_geocent for inj in self.table]
+
+    def pack_data_into_pycbc_format_input(self):
+        data = {}
+        for key in xinj.table[0].__slots__:
+            # Some XML files can have empty columns which are read as None in
+            # python. For these cases we ignore the columns so they do not
+            # appear in the output HDF file. (Such XML files cannot currently
+            # be read by LALSuite C code.) 
+            if getattr(xinj.table[0], key) is not None:
+                data[str(key)] = numpy.array(
+                    [getattr(t, key) for t in xinj.table]
+                )
+
+        for k in ['simulation_id', 'process_id']:
+            a = data.pop(k)
+
+        data['approximant'], data['phase_order'] = numpy.array(
+                [legacy_approximant_name(wf) for wf in data['waveform']]
+        ).T
+        data['tc'] = (
+            data['geocent_end_time'] + 1e-9 * data['geocent_end_time_ns']
+        )
+        data['dec'] = data['latitude']
+        data['ra'] = data['longitude']
+        return data
+
+
+class LVKNewStyleInjectionSet(object):
+    """Reads injections from new-style LVK injection files
+    """
+
+    translation_dict = {
```

Comment:
should this dict rather be defined outside the class?  ie do we gain anything from putting it inside?

Reply:
The dictionary is a property of the new format, so I'd rather keep all of the "code to handle the new format" within the single class. If we have to support multiple input formats here I think it would keep the code cleaner.

Reply:
OK, so it looks like this is a thing called a 'class variable' which is tied to the class, not the instance (`self`). 
I _think_ this requires it to be referenced as `LVKNewStyleInjectionSet.translation_dict`

Reply:
It is indeed a class variable. These *can* be accessed using `self.XXX`:
```
https://www.atatus.com/blog/class-variables-vs-instance-variables-in-java/#accessing-a-class-variable
```
I think a reason not to is doing something like:
```
self.class_variable = 'BLAH'
```
may not make it clear that it would change for more than self!

Here the class variable is a constant, and so I *think* it's okay, and avoids having to write out the class name every time.

The alternative would be to hardcode these things in `__init__` as an instance variable ...

Reply:
OK .. as long as it works as intended